types/src/database_sql.rs

```diff
@@ -1555,6 +1558,74 @@ impl From<BasicWord> for AnnotatedForm {
     }
 }
 
+// Loader to get each chapter id from the database regardless if it's a subchapter or not.
+#[async_trait]
+impl Loader<CollectionChapter> for Database {
+    type Value = Vec<ChapterSingle>;
+    type Error = Arc<sqlx::Error>;
+
+    // string slug can look like "cwkw".chapter1.doc1 But assume user only passes "cwkw"
+
+    async fn load(
+        &self,
+        keys: &[CollectionChapter],
+    ) -> Result<HashMap<CollectionChapter, Self::Value>, Self::Error> {
+        let keys: Vec<_> = keys.iter().map(|k| k.0.clone()).collect();
+        let items = query_file!("queries/collection_chapters.sql", &keys)
+            .fetch_all(&self.client)
+            .await?;
+        Ok(items
+            .into_iter()
+            .map(|chapter| {
+                (
+                    CollectionChapter(chapter.collection_slug),
+                    ChapterSingle {
+                        id: chapter.id,
+                        title: chapter.title,
+                        wordpress_id: chapter.wordpress_id,
+                        index_in_parent: chapter.index_in_parent,
+                        section: if chapter.section == Some("Intro".to_string()) {
+                            CollectionSection::Intro
+                        } else {
+                            CollectionSection::Body
+                        },
```

Comment:
So this isn't the worst way to do it, but we want to be more future proof. The string could change, or we might add new variants. We want this code not to change when we do that. Look at the `impl Loader<PartsOfWord>` and the `word_parts.sql` file for an example where we get sqlx to convert the type for us. Relevant line:

```sql
  word_segment.role as "role: WordSegmentRole"
```

For you, it might be:

```sql
  section as "section: CollectionSection"
```

This tells sqlx that we want this `text` column in the `collection_section` domain to be parsed into the `CollectionSection` rust type.